pkg/config/redisstore.go

```diff
@@ -64,29 +128,56 @@ func newRedisStore(u *url.URL) (rs *redisStore, err error) {
 		}
 	}
 
-	client, err := redis.Dial("tcp", u.Host)
-	if err != nil {
-		return nil, fmt.Errorf("can't connect to the redis server %v: %v", u, err)
-	}
-
 	var password string
 	if u.User != nil {
 		password, _ = u.User.Password()
 	}
 
-	if err := setupConnection(client, password, dbNum); err != nil {
+	client, err := setupRedisConnection(u.Host, password, dbNum, 0)
+	if err != nil {
 		return nil, err
 	}
-	return &redisStore{client, u, 0}, nil
+	rs = &redisStore{
+		client: client,
+		url:    u,
+	}
+	if rs.getKeyspaceAvailability() {
+		// It's better to set the timeout for subscriber client, otherwise client.Recieve()
+		// will block forever. The time is currently static and unconfigurable.
+		sclient, err := setupRedisConnection(u.Host, password, dbNum, redisSubscriberTimeout)
+		if err != nil {
+			glog.Warningf("failed to set up subscriber: %v", err)
+			return rs, nil
+		}
+		sub := &redisSubscriber{
+			client: pubsub.NewSubClient(sclient),
+		}
+		if err := sub.startListening(dbNum); err != nil {
+			glog.Warningf("failed to start subscription: %v", err)
+		} else {
+			rs.redisSubscriber = sub
+		}
+	}
+	return rs, nil
 }
 
 func (rs *redisStore) String() string {
 	return fmt.Sprintf("redisStore: %v", rs.url)
 }
 
+func (rs *redisStore) index(increase int) int {
+	if rs.redisSubscriber == nil {
+		return indexNotSupported
+	}
+	rs.redisSubscriber.mu.Lock()
+	idx := rs.redisSubscriber.lastIndex + increase
```

Comment:
Where is the persistent index updated?
This looks like `rs.redisSubscriber.lastIndex` is never modified.